active/0004-extensibility-support-for-deployment-stacks.md

```diff
@@ -0,0 +1,405 @@
+extension resources with Deployment stacks.
+
+## Terms and definitions
+
+- `Control plane`: An Azure or non-Azure API for managing the lifecycle of resources. This is the API for provisioning
+  or deleting resources.
+- `Extension resource`: An Azure data-plane resource or non-Azure resource declared in a Bicep file or an ARM template.
+- `Extension configuration`: Data needed to connect to a non-Azure control plane to manage resources. This can contain 
+secrets, such as a Kube config for AKS clusters.
+- `Deployment stack`/`stack`: An atomic unit for lifecycle management of a set of resources deployed by a Bicep or ARM template 
+deployment. Resources can be Azure resources or extension resources. Stacks can unmanage resources which results in
+either their deletion or detachment from the stack.
+
+## Motivation
+
+Currently, Deployment stacks do not support managing extension resources deployed through one. Users that utilize
+Deployment stacks will want to manage the lifecycle of any resource deployed through a stack regardless of its control
+plane as this is a key feature of stacks. 
+
+In order to minimally support deletion of extension resources which is the goal of this integration, changes need to be
+made at both the Bicep and ARM template levels to how extension configurations are defined so the Deployment stack 
+service can retrieve necessary secret data in a secure and direct way.
+
+The current implementation of extension configurations in Bicep and ARM templates is too flexible. For example, it 
+allows users to consume template parameters and use template language expressions to manipulate data. This might sound
+great from a user experience perspective, but this creates a problem for the stacks service because it must be able to
+reevaluate the extension configuration at a later time when resource deletion occurs. Non-stack deployments do not have 
+this problem because the configuration is only needed at the time of deployment. The extension configuration can change
+over time due to credential rotations or other reasons that are opaque to the Stacks service. Therefore, it is necessary
+to constrain extension configurations to use constants or simple directives the stacks service can act upon.
+
+## Detailed design
+
+> The core of the REP. Elaborate on the design and implementation with sufficient detail for someone familiar with Bicep
+to grasp. Provide detailed examples to illustrate how the feature is used and its implications on user experience.
+
+The key design change is to move where extension configurations are defined in Bicep and ARM templates. Currently,
+the extension configurations are defined within the deployment of the extension resources being deployed. This REP's 
+proposal is to move these configuration definitions to a dedicated deployment input property that is separate from 
+deployment parameters. See the examples section for Bicep and ARM template examples.
```

Comment:
This is a huge improvement. Extension config should be first-class and part of the deployment resource rather than the template. We implemented this in the Radius fork of the DE and have been living with it happily for some time.